### ISSUE TITLE
[accessibility] Collapsed attribution button should have discernible text

### ISSUE THREAD

**Issue by jnapolitan** [mapbox/mapbox-gl-js#13143]
Hi team,

It seems as though when rendering the collapsed attribution button, the `<span>` nested within the button has an appropriate `title` attribute applied to it, but the `<button>` itself does not. This does not meet WCAG and will fail when running Lighthouse/Axe.
 
![image](https://github.com/mapbox/mapbox-gl-js/assets/13486895/7ba31002-0a2f-45f3-af9e-f14956c0c12f)

**mapbox-gl-js version**: 3.2.0

**browser**: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36

### Steps to Trigger Behavior

 1. Load the map at a screen width that renders the collapsed attribution button
 2. Run Lighthouse/Axe devtools, or inspect the element to see the `<button>` element doesn't have discernible text

**Comment by Malvoz** [comment]
Not sure why `aria-hidden` is used on the `<span>` in this context:

![mapbox-attrib-aria-hidden](https://github.com/user-attachments/assets/ef5e83da-58c2-4f61-9c71-61ea3358ed30)


If `aria-hidden` is removed, the `<button>`  correctly derives the accessible name from the `title`:

![mapbox-attrib-no-aria-hidden](https://github.com/user-attachments/assets/b2aae305-292e-40be-b34c-948d8c88d8f7)

As a side note, I think it'd be more logical to have `title` set on the `<button>` rather than the `<span>`.
